tests/test_filter_jobs.py

```diff
@@ -0,0 +1,21 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.treeherder import TreeherderPage
+
+
+@pytest.mark.nondestructive
+def test_filter_jobs(base_url, selenium):
+    """Open resultset page and filter for platform"""
+    page = TreeherderPage(selenium, base_url).open()
+    platform = u'Linux'
+
+    page.filter_by(platform)
+    assert platform in page.result_sets[0].builds[0].platform_name
+
+    platform2 = u'Windows'
+    page.filter_by(platform2)
+    assert platform not in page.result_sets[0].builds[0].platform_name
```

Comment:
Clarification: It isn't easy to make a confirmation of the state of the filter before searching as there are too many options to confirm each one. So I went with one positive and one negative assert to cover different scenarios.